pkg/azurefile/azurefile.go

```diff
@@ -876,6 +887,78 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
 	})
 }
 
+// CopyFileShare copies a fileshare in the same storage account
+func (d *Driver) copyFileShare(ctx context.Context, req *csi.CreateVolumeRequest, accountKey string, shareOptions *fileclient.ShareOptions, storageEndpointSuffix string) error {
+	if shareOptions.Protocol == storage.EnabledProtocolsNFS {
+		return fmt.Errorf("protocol nfs is not supported for volume cloning")
+	}
+	var volumeID string
+	if req.GetVolumeContentSource() != nil && req.GetVolumeContentSource().GetVolume() != nil {
+		volumeID = req.GetVolumeContentSource().GetVolume().GetVolumeId()
+	}
+	resourceGroupName, accountName, srcFileShareName, _, _, _, err := GetFileShareInfo(volumeID) //nolint:dogsled
+	dstFileShareName := shareOptions.Name
+	if err != nil {
+		return status.Error(codes.NotFound, err.Error())
+	}
+	klog.V(2).Infof("begin to copy fileshare %s to %s", srcFileShareName, dstFileShareName)
+	if srcFileShareName == "" {
+		return fmt.Errorf("srcFileShareName is empty")
+	}
+	if dstFileShareName == "" {
+		return fmt.Errorf("dstFileShareName is empty")
+	}
+
+	klog.V(2).Infof("generate sas token for account(%s)", accountName)
+	accountSasToken, genErr := generateSASToken(accountName, accountKey, storageEndpointSuffix, d.sasTokenExpirationMinutes)
+	if genErr != nil {
+		return genErr
+	}
+
+	timeAfter := time.After(waitForCopyTimeout)
+	timeTick := time.Tick(waitForCopyInterval)
+	srcPath := fmt.Sprintf("https://%s.file.%s/%s%s", accountName, storageEndpointSuffix, srcFileShareName, accountSasToken)
+	dstPath := fmt.Sprintf("https://%s.file.%s/%s%s", accountName, storageEndpointSuffix, dstFileShareName, accountSasToken)
+
+	jobState, percent, err := getAzcopyJob(dstFileShareName)
```

Comment:
this logic exists 3 times in the code:
```
	jobState, percent, err := getAzcopyJob(dstFileShareName)
	if err != nil || jobState == AzcopyJobError {
		return fmt.Errorf("failed to get azcopy job with error: %v, jobState: %v", err, jobState)
	}
	if jobState == AzcopyJobCompleted {
		klog.V(2).Infof("azcopy job is completed, copied fileshare %s to %s successfully", srcFileShareName, dstFileShareName)
		return nil
	} else if jobState == AzcopyJobRunning {
		klog.V(2).Infof("azcopy job is in progress, copy percent: %s%", percent)
	}
```

could you refine the code a little, e.g. use a new function, that would make the logic more clear.

Reply:
add logs and warnings into `getAzcopyJob` func and improve this logic.